File: greedy/leetcode_2500/solution.js
/**
 * @param {number[][]} grid
 * @return {number}
 */
var deleteGreatestValue = function(grid) {
    let result = 0;

    const m = grid.length;
    const n = grid[0].length;

    for(let i=0; i<m; i++) {
        grid[i].sort((a, b) => b - a);
    }

    for (let j=0; j<n; j++) {
        let maxValue = 0;

        for (let i=0; i<m; i++) {
            maxValue = Math.max(maxValue, grid[i][j])
        }

        result += maxValue
    }

    return result;
};

File: greedy/leetcode_2500/solution.py
from typing import List

class Solution:
    def deleteGreatestValue(self, grid: List[List[int]]) -> int:
        result = 0

        m = len(grid)
        n = len(grid[0])

        for i in range(m):
            grid[i].sort(reverse=True)

        for j in range(n):
            max_value = 0

            for i in range(m):
                max_value = max(max_value, grid[i][j])

            result += max_value

        return result